react-common/components/share/ShareInfo.tsx

```diff
@@ -167,6 +169,12 @@ export const ShareInfo = (props: ShareInfoProps) => {
 
     const inputTitle = prePublish ? lf("Project Title") : lf("Project Link")
 
+    if (isLoggedIn) {
+        pxt.tickEvent("share.open.loggedIn", { state: shareState, anonymous: isAnonymous.toString(), persistent: hasProjectBeenPersistentShared.toString() });
+    } else {
+        pxt.tickEvent("share.open", { state: shareState});
+    }
+
```

Comment:
How do we capture the state if this is called only once in the beginning?

Reply:
I see, if the intention is for this tick to run on any transition in the share dialog then we can watch for those with

```suggestion
    React.useEffect(() => {
        if (isLoggedIn) {
            pxt.tickEvent("share.open.loggedIn", { state: shareState, anonymous: isAnonymous.toString(), persistent: hasProjectBeenPersistentShared.toString() });
        } else {
            pxt.tickEvent("share.open", { state: shareState});
        }
    }, [state, isAnonymous, hasProjectBeenPersistentShared]);
```

and it will rerun when any of those 3 values are changed